engine/engine.go

```diff
@@ -211,14 +209,40 @@ func (e Engine) launchResolution(publicID string, async bool, sm *semaphore.Weig
 
 	res.Values.SetConfig(e.config)
 
-	// all ready, run remaining steps
-
-	utask.AcquireExecutionSlot()
-
+	// checking if all resources are available before starting the resolution
+	// first, checking if we have a custom semaphore, for example, crashed instance recover
+	// putting this semaphore first, cause the crashed instance limitation is always lower than the execution pool
 	if sm != nil {
-		sm.Acquire(context.Background(), 1)
+		if err := sm.Acquire(shutdownCtx, 1); err != nil {
+			debugLogger.Debugf("Engine: launchResolution() %s acquire resource: instance is shutting down", res.PublicID)
+			return nil, errors.New("instance is shutting down")
+		}
+	}
+	// second, checking if we have restricted resource on the given template
+	// template could be completely desactivated, as a "dead resource", if it's the case, we need to exit because
+	// the limit won't change until instance rebooted
```

Comment:
```suggestion
	// second, check if we have a restricted resource on the given template
	// template could be completely deactivated as a "dead resource". If it's the case, we need to exit because
	// the limit won't change until next instance's reboot
```